jstests/auth/cluster_ip_allowlist_replset.js

```diff
@@ -25,38 +25,38 @@ admin.auth('admin', 'admin');
 assert.commandWorked(primary.adminCommand(
     {setDefaultRWConcern: 1, defaultWriteConcern: {w: "majority"}, writeConcern: {w: "majority"}}));
 
-jsTest.log("ReplSet started, will block __system connections and expect error");
-
-assert.commandWorked(
-    primary.adminCommand({setParameter: 1, "clusterIpSourceAllowlist": ["192.0.2.1"]}));
-
 function resetClusterIpSourceAllowlist(host) {
     jsTest.log("resetClusterIpSourceAllowlist: started");
-    sleep(15000);
     const mongo = new Mongo(host);
     const admin = mongo.getDB("admin");
     admin.auth('admin', 'admin');
+
+    jsTest.log("Check log for denied connections");
+    checkLog.containsJson(admin, 20240, {});
+    jsTest.log("Found denied connection(s)");
+
     assert.commandWorked(admin.adminCommand({setParameter: 1, "clusterIpSourceAllowlist": null}));
     jsTest.log("Successfully reset clusterIpSourceAllowlist");
 }
 
 let thread = new Thread(resetClusterIpSourceAllowlist, primary.host);
 thread.start();
 
-jsTest.log("Attempt to add host. This should be failing, then succeed");
-
 const arbiterConn = replTest.add();
 const conf = replTest.getReplSetConfigFromNode();
 conf.members.push({_id: 2, host: arbiterConn.host, arbiterOnly: true});
 conf.version++;
 
 // Following function will succeed after resetClusterIpSourceAllowlist()
 // successfully resets IP for connection
+jsTest.log("ReplSet started, will block __system connections and expect error");
+assert.commandWorked(
+    primary.adminCommand({setParameter: 1, "clusterIpSourceAllowlist": ["192.0.2.1"]}));
 assert.commandWorked(admin.runCommand({replSetReconfig: conf}));
 thread.join();
 
 jsTest.log("Verify that connections were denied");
-checkLog.containsJson(admin, 20240, {}, 15000);
+checkLog.containsJson(admin, 20240, {}, 1);
```

Comment:
Curious if this is even necessary in the end, since the threaded function also has this call. Can an assert failure from a thread not trigger a test failure?

Reply:
I verified that an error does propagate, but it looks like threading code in tests is not very mature and the way it works could [inadvertently] change in future. This line is to ensure test fails even if an assert does not propagate from `thread.join()`